File: KSTU_studying/c++_basis/lab_2/python/ex03.py
arr = [99, 32, 42, 255, 32, 32 ,55, 66, 21 ,23]

for i in range(len(arr)):
    if i < 5:
        for k in range(int(len(arr) / 2)):
            if arr[i] < arr[k]:
                arr[i], arr[k] = arr[k], arr[i]
    else:
        for k in range(int(len(arr) / 2), len(arr)):
            if arr[i] > arr[k]:
                arr[i], arr[k] = arr[k], arr[i]

for i in range(len(arr)):
    if i < len(arr) - 1:
        print(arr[i], end=",")
    else:
        print(arr[i], end="")

File: KSTU_studying/c++_basis/lab_2/c++/ex03.cpp
#include <iostream>
#include <vector>

int main(void) {
    int array_count = 10;
    std::vector<int>arr(array_count);
    for (int i = 0; i < array_count; i++) {
        arr[i] = rand() % 11 - 5;
    }
    //список
    std::cout << "array:" << "\n";
    for (int i = 0; i < array_count; i++) {
        std::cout << "[" << arr[i] << "]";
    }
    //сортированный список
    int check = 1;
    while(check == 1) {
        check = 0;
        for (int i = 0; i < array_count / 2 - 1; i++) {
            if (arr[i] > arr[i + 1]) {
                std::swap(arr[i], arr[i + 1]);
                check = 1;
            }
        }
    }
    check = 1;
    while(check == 1) {
        check = 0;
        for (int i = array_count / 2; i < array_count - 1; i++) {
            if (arr[i] < arr[i + 1]) {
                std::swap(arr[i], arr[i + 1]);
                check = 1;
            }
        }
    }
    std::cout << "\n" << "sorted array:" << "\n";
    for (int i = 0; i < array_count; i++) {
        std::cout << "[" << arr[i] << "]";
    }
}